xarray/backends/api.py

```diff
@@ -1642,38 +1761,57 @@ def open_mfdataset(
 
     # Combine all datasets, closing them in case of a ValueError
     try:
+        # for any data_vars, coords, compat, or join that is  None, set to
+        # default - with warning if appropriate
+        kwargs_with_defaults = _get_default_open_mfdataset_kwargs(
+            datasets,
+            concat_dim=concat_dim,
+            data_vars=data_vars,
+            coords=coords,
+            compat=compat,
+            join=join,
+        )
         if combine == "nested":
             # Combined nested list by successive concat and merge operations
             # along each dimension, using structure given by "ids"
             combined = _nested_combine(
                 datasets,
                 concat_dims=concat_dim,
-                compat=compat,
-                data_vars=data_vars,
-                coords=coords,
+                **kwargs_with_defaults,
                 ids=ids,
-                join=join,
                 combine_attrs=combine_attrs,
             )
         elif combine == "by_coords":
             # Redo ordering from coordinates, ignoring how they were ordered
             # previously
             combined = combine_by_coords(
                 datasets,
-                compat=compat,
-                data_vars=data_vars,
-                coords=coords,
-                join=join,
+                **kwargs_with_defaults,
                 combine_attrs=combine_attrs,
             )
         else:
             raise ValueError(
                 f"{combine} is an invalid option for the keyword argument ``combine``"
             )
-    except ValueError:
+    except ValueError as e:
         for ds in datasets:
             ds.close()
-        raise
+        # When using new defaults, add a warning if the failure appears to be related
+        # to the new kwarg values.
+        for kwarg, new, old in (
+            (data_vars, "data_vars='minimal'", "date_vars='all'"),
+            (coords, "coords='minimal'", "coords='different'"),
+            (compat, "compat='override'", "compat='no_conflicts'"),
+            (join, "join='exact'", "join='outer'"),
+        ):
+            if kwarg is None and new in str(e):
+                warnings.warn(
+                    f"Failure might be related to new default ({new}) "
+                    f"in `open_mfdataset`. Previously the default was {old}",
+                    category=UserWarning,
+                    stacklevel=2,
+                )
```

Comment:
These are the warnings that are meant to provide extra context on failures when you set `use_new_open_mfdataset_kwargs=True` and for a while after xarray switches to using the new defaults by default.